src/api/user_presence.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::clock::ClockInt;
+use crate::ctap::Channel;
+use embedded_time::duration::Milliseconds;
+
+pub enum UserPresenceError {
+    Declined,
+    Canceled,
+    Timeout,
+}
+
+pub type UserPresenceResult = Result<(), UserPresenceError>;
+
+pub trait UserPresence {
+    /// Called at the beginning of user presence checking process.
```

Comment:
nit: documentation should describe what the function does/should do, not when it is called.

```rust
/// Initializes for a new user presence check.
///
/// Must eventually be followed by a call to `Self::check_complete`.
```

Reply:
Done